RFCs/0011-Discover-Only-TestProjects.md

```diff
@@ -0,0 +1,76 @@
+# 0011 Discover for Only Test Projects
+
+## Summary
+This note outlines the proposed changes to the TestWindow component inside VS IDE,
+for making improvements in the test discovery.
+
+## Motivation
+* Scope of improvement in discovery through TestExplorer. See [here](#Discovery.Proof)
+* Need to standardize the rules for identifying Test projects.
+
+## Problem
+TestWindow component inside VS IDE currently sends all the projects present in the solution for discovery.
+This list of projects includes projects that do not contain any tests.
+Sending redundant projects to the runner creates an overhead. Each adapter will try and fail to discover any tests inside such projects.
+
+### Scenarios
+There are two ways user can author C# based tests
+1. Create a test project using Unit Test template
+2. Add the appropriate adapter or framework references to a class library project. 
+
+### Current State
+There are existing ways to identify if a project is a test project. Brief explanation is as follows.
+
+#### Test project Guid
+All C# based classic Unit tests which are created using the default templates have a special ProjectType guid to be identified as Test project. 
+```
+<ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB}<ProjectTypeGuids>
+```
+
+#### Test Service Guid
+Currently, there is another special guid that we dynamically add to projects that contain tests and were not created via default template.
+How this works is, since all the projects are used for discovery, the projects which actually contain tests are identified post discovery
+operation and a service guid is added to them.
+
+```
+ <ItemGroup>		
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />		
+  </ItemGroup>
+```
+
+#### Test container capability
+The latest and recommended way of getting a project identified as a Test project is adding
+*TestContainer* capability.
+.Net core based test projects created using the template have this capabilty by default.
+
+```
+<ItemGroup>
+    <ProjectCapability Include="TestContainer" />
+</ItemGroup>
+```
+
+## Proposed Changes
+Add an ability to identify the test projects using all the three rules mentioned above.
+Also, provide user an option to run discovery operation for test projects only.
+
+User will be provide with two options,
+1. Discover All projects.
+2. Discover Test projects
```

Comment:
It's not clear to me that this is ever going to be a good idea, because it (presumably silently) blocks you from manually adding unit test support to a new or existing library; it will be ignored unless it contains one of the magic indicators. I personally can't ever imagine setting (or recommending to anybody) "Discover Test Projects". This feels like you're trying to solve a specific problem without naming it, and doing so in such a way that adding new test projects becomes potentially very user-unfriendly.